### PR TITLE
Add mirroring of master to main branch.

### PR DESCRIPTION
**Pull request by godofredoc**
This is to prepare for the migration of master to main.

Bug: https://github.com/flutter/flutter/issues/90476

## Pre-launch Checklist

- [X] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [X] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [X] I read and followed the [relevant style guides] and ran [the auto-formatter]. (Note that unlike the flutter/flutter repo, the flutter/plugins repo does use `dart format`.)
- [X] I signed the [CLA].
- [X] The title of the PR starts with the name of the plugin surrounded by square brackets, e.g. `[shared_preferences]`
- [X] I listed at least one issue that this PR fixes in the description above.
- [X] I [updated pubspec.yaml](https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#version-and-changelog-updates) with an appropriate new version according to the [pub versioning philosophy].
- [X] I updated CHANGELOG.md to add a description of the change.
- [X] I updated/added relevant documentation (doc comments with `///`).
- [X] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test exempt.
- [X] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/plugins/blob/master/CONTRIBUTING.md
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[relevant style guides]: https://github.com/flutter/plugins/blob/master/CONTRIBUTING.md#style
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[the auto-formatter]: https://github.com/flutter/plugins/blob/master/script/tool/README.md#format-code

### CODE REVIEW

**Review by stuartmorgan — APPROVED** [review]
LGTM

File: .github/workflows/mirror.yml [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    branches:
+      - 'master'
+
+jobs:
+  mirror_job:
+    permissions:
+      pull-requests: write
+    runs-on: ubuntu-latest
+    if: ${{ github.repository == 'flutter/plugins' }}
+    name: Mirror main branch to master branch
```
Comment (stuartmorgan) [inline .github/workflows/mirror.yml:17]: Oops, I just noticed that this is backward. That's probably true of the flutter/packages version as well.

Comment (godofredoc) [reply .github/workflows/mirror.yml]: updated the name.

**Review by stuartmorgan — APPROVED** [review]
LGTM

**Pull request merged**